internal/hmyapi/harmony.go

```diff
@@ -65,3 +66,21 @@ func (s *PublicHarmonyAPI) GetNodeMetadata() NodeMetadata {
 		s.b.GetShardID(),
 	}
 }
+
+// GetCurrentTransactionErrorSink ..
+func (s *PublicHarmonyAPI) GetCurrentTransactionErrorSink() []core.StakingTransactionError {
+	result := []core.StakingTransactionError{}
+	count := 0
+	core.StakingTransactionErrorSink.Range(func(key, value interface{}) bool {
+		count++
+		result = append(result, value.(core.StakingTransactionError))
+		return true
+	})
+	const maxMapSize = 1024
+	if count == maxMapSize {
```

Comment:
This logic is confusing to me. Something is wrong.

Reply:
If I reached the max size that I wanted to keep in memory, then delete it all

Reply:
alternative i thought of using is a circular queue, or something circular data structure